internal/gapicgen/generator/gapics.go

```diff
@@ -266,14 +266,42 @@ func (g *GapicGenerator) microgen(conf *microgenConfig) error {
 
 // manifestEntry is used for JSON marshaling in manifest.
 type manifestEntry struct {
-	DistributionName  string `json:"distribution_name"`
-	Description       string `json:"description"`
-	Language          string `json:"language"`
-	ClientLibraryType string `json:"client_library_type"`
-	DocsURL           string `json:"docs_url"`
-	ReleaseLevel      string `json:"release_level"`
+	DistributionName  string      `json:"distribution_name"`
+	Description       string      `json:"description"`
+	Language          string      `json:"language"`
+	ClientLibraryType string      `json:"client_library_type"`
+	DocsURL           string      `json:"docs_url"`
+	ReleaseLevel      string      `json:"release_level"`
+	LibraryType       LibraryType `json:"library_type"`
 }
 
+type LibraryType int
+
+func (l LibraryType) MarshalJSON() ([]byte, error) {
+	switch l {
+	case GapicAutoLibraryType:
+		return []byte(`"GAPIC_AUTO"`), nil
+	case GapicManualLibraryType:
+		return []byte(`"GAPIC_MANUAL"`), nil
+	case CoreLibraryType:
+		return []byte(`"CORE"`), nil
+	case AgentLibraryType:
+		return []byte(`"AGENT"`), nil
+	case OtherLibraryType:
+		return []byte(`"OTHER"`), nil
+	default:
+		return nil, fmt.Errorf("unable to marshal value: %v", l)
+	}
+}
+
+const (
+	GapicAutoLibraryType LibraryType = iota
```

Comment:
Could we just make these strings to avoid needing `MarshalJSON` at all?

Reply:
That is a great point. Done